pkg/daemon/discover/discover.go

```diff
@@ -167,9 +172,27 @@ func udevBlockMonitor(c chan string, period time.Duration) {
 	// return any add or remove events, but none that match device mapper
 	// events. string matching is case-insensitve
 	events := make(chan string)
+
+	// get discoverDaemonUdevBlacklist from the enviornment variable
+	// if user doesnt provide any regex; generate the default regex;
+	// if user provide nil value; then use no regex values
+	// else use the regex provided by user
+	discoverUdev := os.Getenv(discoverDaemonUdev)
+	if discoverUdev == "" {
+		udevFilter = []string{"(?i)dm-[0-9]+", "(?i)rbd[0-9]+", "(?i)nbd[0-9]+"}
+		logger.Infof("using the default regular expressions %q", udevFilter)
+	} else if discoverUdev == "nil" {
```

Comment:
I don't think this case exists, no one is going to set `DISCOVER_DAEMON_UDEV_BLACKLIST=nil`.

Reply:
Only `discoverUdev == ""` is valid.

Reply:
@BlaineEXE raised the concern that what if the user wants no disk to blacklisted. A rare occurrence ofcourse but still a possibility right. That's why I put in the case nil if the user wants nothing to be blacklisted. I commented this in the operator yaml files stating the nil case.

Reply:
Then I'd suggest not to apply any filters at all from the beginning. So if `DISCOVER_DAEMON_UDEV_BLACKLIST` is not set let's not apply any filters. If `DISCOVER_DAEMON_UDEV_BLACKLIST` is set, honor what was requested by the admin.

Also, one thing to consider is that c-v will soon be used by the discover pod with https://github.com/rook/rook/pull/4053, and c-v has filters built-in already so this might not take effects at all. Basically, your filter might be applied and the device selected but it'll be rejected by c-v.

Reply:
We need a default filter. During upgrades this property won't be set, so unfortunately we need a built-in default.

I believe #4053 doesn't change this filtering behavior. It's only adding a new call to c-v and leaving the existing device discovery the same, right?

I'd vote not to have the "nil" option.

Reply:
Why don't we set `DISCOVER_DAEMON_UDEV_BLACKLIST` with the filters we want into the operator.yaml? And add a manual step in the upgrade procedure to add it?

With that, we get the flexibility we want I think.

Reply:
Along the upgrade thought, can we have this added automatically with the Helm charts as well? I feel like I should know more about how Helm works...

Reply:
> I want to point out that we are talking about a Rook generic daemon with a purely Ceph-oriented viewpoint. We need to make sure we are considering the needs of other Rook backends, and I don't think we can make many assumptions about how other backends would like to behave.

So, in that case, its best to leave it to user to set the variable, right? I mean if it needs to be fully configurable then the user should be able to input any value he can. I would also like to keep the default because what if the user is not so advanced and he would like to go with what is already being provided. I think that needs to be kept.

Reply:
If my knowledge is correct, the if the user left the regex as empty then it will throw some error right? @travisn @leseb @BlaineEXE

Reply:
It would be great to have a unit test that checks the behavior of an empty or invalid regex. But I don't think the regex will ever be empty. I'm voting for a default unless overridden from the env var.

Reply:
We do need to have unit tests for this additional functionality, and those should include error cases.